File: C_Programming/Study/Q3_Assignment12.c

///////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name :  CountTwo
// Output:          Count frequency of 2 in that number
// Description:     Assignment_12: Q.3  : Write a program which accept Number from user & Count frequency of 2
//                                        in it.
// Date:            30/05/2026
// Author:          Shital Ajit Nikam   
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////

#include <stdio.h>

int CountTwo(int iNo)
{
    int iDigit = 0;
    int iCnt = 0;

    //Updator used
    if(iNo < 0)
    {
        iNo = -iNo;
    }
               
    while(iNo != 0)
    {
        iDigit = iNo % 10;
        
        if(iDigit == 2)
        {
            iCnt++;
        }
            
        iNo = iNo /10;
                
    }
    return iCnt;
}

int main()
{
    int iValue = 0;
    int iRet = 0;
        
    printf("Enter Number: \n");
    scanf("%d",&iValue);

    iRet = CountTwo(iValue);

    printf("Count frequency of 2 is : %d",iRet);
    
    return 0;
}

/////////////////////////////////////////////////////////////////////////////////////
//Time Complexity : O(N)  where N = length of Number
/////////////////////////////////////////////////////////////////////////////////////
// Input : 2395
// Output : Count Frequncy of 2 is : 1
// Input : 1018
// Output : Count Frequncy of 2 is : 0
// Input : 9000
// Output : Count Frequncy of 2 is : 0
// Input : 9222
// Output :Count Frequncy of 2 is : 3
// 
/////////////////////////////////////////////////////////////////////////////////////

File: Python_Programming/Competative/Q3_Assignment12.py
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Function Name :  Add , Sub, Mult, Div
#        
# Description:     Assignment_12: Q.3: Write a program which accepts two Numbers & print its Addition,Substractio, 
#                                      Multiplication,Division. 
#                                     
# Date:            30/06/2026
# Author:          Shital Ajit Nikam   
#
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////


# Lambda function defination

Add = lambda No1,No2 : No1 + No2

Sub = lambda No1,No2 : No1 - No2

Mult = lambda No1,No2 : No1 * No2

Div = lambda No1,No2 : No1 // No2                   # Integer division needs // operator
    
def main():
    value1 = int(input("Enter the First Number: "))
    value2 = int(input("Enter the Second Number: "))
    
    Ret = Add(value1,value2)
    print("Addition is :",Ret)
    
    Ret = Sub(value1,value2)
    print("Substraction is :",Ret)
    
    Ret = Mult(value1,value2)
    print("Multiplication is :",Ret)
    
    Ret = Div(value1,value2)
    print("Division is :",Ret)       
              
    
if __name__ == "__main__":
    main()